File: AlienStudio/src/app/models/UI/IForm.ts
import {IXMLBase} from '../IXMLBase';
import {ILoadDetails} from './ILoadDetails';

export class IForm extends IXMLBase {
  control: any;
  type = 'form';
  entity: string;
  name: string;
  loadDetails: ILoadDetails


  constructor(control: any, entity: string, name: string, loadDetails: ILoadDetails) {
    super();
    this.control = control;
    this.entity = entity;
    this.name = name;
    this.loadDetails = this.loadDetails;
  }
}

File: connectionhub/Utils/src/main/java/com/business/utils/models/UI/IForm.java
package com.business.utils.models.UI;

import com.business.utils.models.IXMLBase;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlRootElement;

@JsonPropertyOrder({"loadDetails","control"})
@JacksonXmlRootElement(localName = "form")
public class IForm extends IXMLBase {
    @JacksonXmlProperty(isAttribute = true)
    String name;
    @JacksonXmlProperty(isAttribute = true)
    String entity;
    String type;
    IView control;
    @JacksonXmlProperty(localName = "loadDetails")
    ILoadDetails loadDetails;

    public IForm() {
    }

    @JsonIgnore()
    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public IView getControl() {
        return control;
    }

    public void setControl(IView control) {
        this.control = control;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getEntity() {
        return entity;
    }

    public void setEntity(String entity) {
        this.entity = entity;
    }

}
